dagobah/core/delegator.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+from ..backend.base import BaseBackend
+
+logger = logging.getLogger('dagobah')
+
+class CommitDelegator(object):
+    """
+    The CommitDelegator is a class designed to handle any and all commits
+    to the database.
+
+    The delegator takes in commit requests from various objects, and decides
+    if they should actually be committed, and if so, manage any special
+    processing.
+    """
+
+    def __init__(self, backend=BaseBackend()):
+        self.backend = backend
+
+    def commit_dagobah(self, dagobah, cascade=False):
+        """ Commit this Dagobah instance to the backend.
+
+        If cascade is True, all child Jobs are commited as well.
+        """
+        logger.debug('Committing Dagobah instance with cascade={0}'.
+                     format(cascade))
+        self.backend.commit_dagobah(dagobah._serialize())
+        if cascade:
+            [self.commit_job(job) for job in dagobah.jobs]
```

Comment:
yeah, this is old code that I just ported over to the class (see dagobah.py L122 in the changes, being deleted by this PR). Dagobah has always been a bit excessive. Might be a good time for refactoring though.